### PR TITLE
Fix issues introduced in the feature_find_settings merge

### PR DESCRIPTION
**Pull request by fieldOfView**
This PR fixes two minor issues introduced in the feature_find_settings merge:
* When starting Cura in simple mode, the filter toggle is incorrectly showing up on the "advanced" tab
* A qml error about anchoring

### CODE REVIEW

File: resources/qml/Sidebar.qml [hunk 1]
```diff
@@ -166,15 +166,6 @@ Rectangle
...
 
-    currentModeIndex:
-    {
-        var index = parseInt(UM.Preferences.getValue("cura/active_mode"))
-        if(index)
-        {
-            return index;
-        }
-        return 0;
-    }
```
Comment (Ghostkeeper) [inline resources/qml/Sidebar.qml]: Wait, but this is still being used at (new) line 262, where you added additional checks?

Comment (fieldOfView) [inline resources/qml/Sidebar.qml]: Yes. The important bit of this code is moved to Component.onCompleted.
The problem I'm fixing here is that currentModeIndex was being set before the modesListModel was populated.

**Comment by Ghostkeeper** [comment]
Still getting these QML errors that might be caused by a filter such as this:

    qml: TableViewSelection: index out of range
    qml: TableViewSelection: index out of range

But I can't attribute them to this feature. One of my largest gripes with QML is that the error reporting is so bad. There are no more problems with anchoring though, and it doesn't display the magnifying glass on start-up in recommended mode. Still seems to work fine for the rest. Thanks for the fix!

**Comment by fieldOfView** [comment]
I've been trying to track down these TableViewSelection notices since before the 2.3 beta

**Comment by awhiemstra** [comment]
Those messages come from the Preferences dialog and started showing up iirc when changing from Qt 5.4 to Qt 5.5.

**Pull request merged**